client/src/message_interface/message_handler.rs

```diff
@@ -670,6 +671,9 @@ impl ClientMessageHandler {
                 token_scheme_kind,
             ))),
             Message::Faucet { url, address } => Ok(Response::Faucet(request_funds_from_faucet(&url, &address).await?)),
+            Message::HashTransactionEssence { essence } => Ok(Response::TransactionEssenceHash(prefix_hex::encode(
+                TransactionEssence::try_from_dto_unverified(&essence)?.hash(),
```

Comment:
I know user is just asking for a hash, should we still validate it against the currently use network?

Reply:
I wouldn't do it, because then you can't use it offline (unless you provide the network id also there, which just makes it more complicated to use)

Reply:
Fair enough, wasn't convinced either